### PR TITLE
Resolve still-present column overflow issues

### PR DESCRIPTION
**Pull request by jernestmyers**
Resolves https://github.com/VEuPathDB/WDKClient/issues/285

It seems the `useIsRefOverflowingHorizontally` gets out of sync and thus its use as a condition in the truncation logic is problematic. Removing it seems to behave as desired, leaving only this check to determine truncation:
```js
if (
  ref.current.innerText.length > attribute.truncateTo &&
  isEqual(styleSpec, defaultStyleSpec) 
)
```

### CODE REVIEW

**Review by aurreco-uga — APPROVED** [review]

**Comment by dmfalke** [comment]
I still see overflow issues when I change the sort column.

To reproduce in PlasmoDB:
1. Search genes with user comments
2. When the result table loads, change the sort column to **Comment Headlines**.
3. Some cells in **Genomic Location (gene)** will have content overflowing into the next cell.

I think this is happening because of how Mesa calculates column widths. Mesa doesn't recalculate column widths when the table is sorted. This means that the content in a cell can become wider than the width allotted, but its length may not exceed the length the atrribute's `truncateTo` value.

If we revert the changes in this PR and address the `FIXME` in https://github.com/VEuPathDB/WDKClient/blob/a0b8506b46f8d9a7ce4fb2045024281fbec413ff/Client/src/Hooks/Overflow.ts#L12-L19, I think it will fix this issue.

However, I'm starting to think that Mesa should support truncation and implement the overflow logic currently in `AttributeCell`. This could be done by adding a `truncateTo` property to Mesa columns. I don't think we should do this now, until we have another need for truncating table columns.

I think Mesa's behavior with column widths is desirable, in that sorting does not cause the table dimensions to change.

In short, I think we should keep the overflow check and try to fix the issue in `useIsRefOverflowing`.

**Comment by jernestmyers** [comment]
Okay, I notice the `ResizeOberserver API` is now compatible with all browsers.

Thanks for catching the still-present truncation issues. I was only looking at the column that I sorted on, which was no longer overflowing. Didn't think neighboring columns would be affected. I initially forced column width calculations when the sorting changed, which didn't fix the truncation issue itself, but perhaps in tandem it would have.

Either way, I'll try to beef up that hook.

**Review by dmfalke — APPROVED** [review]
